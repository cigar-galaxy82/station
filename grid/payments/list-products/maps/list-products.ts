/* eslint-disable jest/no-export */

import { describe, expect } from '@jest/globals';
import { SuperfaceTest } from '@superfaceai/testing';

export function listProductsTest(providerName: string): void {
  describe(`payments/list-products/${providerName}`, () => {
    let superface: SuperfaceTest;

    beforeEach(() => {
      superface = new SuperfaceTest({
        profile: 'payments/list-products',
        provider: providerName,
        testInstance: expect,
      });
    });

    describe('ListProducts', () => {
      describe('when all inputs are correct', () => {
        it('creates a product', async () => {
          const result = await superface.run({
            useCase: 'ListProducts',
            input: {},
          });
          expect(() => result.unwrap()).not.toThrow();
          expect(result.unwrap()).toMatchSnapshot();
        });
      });
    });
  });
}
