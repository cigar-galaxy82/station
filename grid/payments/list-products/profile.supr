name = "payments/list-products"
version = "1.0.0"

"""
List products
"""

usecase ListProducts safe {
  result [
    {
      """
      Product ID
      Unique identifier of the product the plan is for (if applicable).
      """
      id! string!

      """
      Name
      Name of the product.
      """
      name string!

      """
      Description
      Description of the product.
      """
      description string!

      """
      Product type
      Indicates whether the product is physical or intangible (e.g. a service, or digital goods).

      Possible values:
        - `physical`
        - `digital`
        - `service`
      """
      type enum {
        physical,
        digital,
        service
      }
    }
  ]
  
  error {
    """
    Title
    A short, human-readable summary of the problem type.
    """
    title! string

    """
    Detail
    A human-readable explanation specific to this occurrence of the problem.
    """
    detail string
  }

  example Successful {
    result [
      {
        id = "PROD-123908"
        name = "Superface milk"
        description = "Milk that gives you super-powers, delivered to your door."
        type = "physical"
      },
      {
        id = "PROD-123909"
        name = "Superface milk price monitor"
        description = "Gives you the most up-to-date info on milk prices."
        type = "service"
      },
      {
        id = "PROD-123910"
        name = "Superface theme song"
        description = "Our theme song in MP3 and FLAC format. (Features laser sharks)"
        type = "digital"
      }
    ]
  }
}