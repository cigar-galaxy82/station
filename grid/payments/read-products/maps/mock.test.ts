import { listProductsTest } from './list-products';

listProductsTest('mock');
