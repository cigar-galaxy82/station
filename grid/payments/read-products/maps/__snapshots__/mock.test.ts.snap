// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payments/read-products/mock ListProducts when all inputs are correct creates a product 1`] = `
Array [
  Object {
    "description": "Milk that gives you super-powers, delivered to your door.",
    "id": "PROD-123908",
    "name": "Superface milk",
    "type": "physical",
  },
  Object {
    "description": "Gives you the most up-to-date info on milk prices.",
    "id": "PROD-123909",
    "name": "Superface milk price monitor",
    "type": "service",
  },
  Object {
    "description": "Our theme song in MP3 and FLAC format. (Features laser sharks)",
    "id": "PROD-123910",
    "name": "Superface theme song",
    "type": "digital",
  },
]
`;
